addons/web/static/src/views/view_button/view_button_hook.js

```diff
@@ -56,7 +56,7 @@ export function useViewButtons(model, ref, options = {}) {
                 let buttonContext = {};
                 if (clickParams.context) {
                     if (typeof clickParams.context === "string") {
-                        const valuesForEval = Object.assign({}, record.data, {
+                        const valuesForEval = Object.assign({}, record.evalContext, {
                             active_id: resId,
```

Comment:
check if active_ids and all those are not outdated by https://github.com/odoo/odoo/pull/96910